roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -88,6 +88,38 @@ object tuple {
     }
   }
 
+  /**
+    * Define the "tuple-safe-nth" primitive.
+    * This safely returns the nth element of the specified Tuple
+    * e.g. (tuple-safe-nth [1 2 3 4 5] 3) ==> 4
+    */
+  object tplSafeNth extends Prim {
+    override val name: String = "tuple-safe-nth"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RFixnum] = {
+      val elem = ctxt.argvec.elem
+
+      checkTuple(0, elem).map { t =>
+        checkRFixnum(1, elem) match {
```

Comment:
You would do a `map` here instead of a `match` because you are only interested in the `Right` case.

Reply:
Done